### ISSUE TITLE
Building for real iOS Device fails, Simulator works fine

### ISSUE THREAD

**Issue by maximilianmaihoefner** [invertase/firestore-ios-sdk-frameworks#24]
Hi,

I have been trying to build my Flutter App to test it on a real iOS Device. But I cannot figure out how to resolve the error below, while still including the pre-compiled binaries:
`
ld: in /path/to/app/build/ios/Debug-iphoneos/BoringSSL-GRPC.framework/BoringSSL-GRPC(BoringSSL-GRPC-dummy.o), building for iOS, but linking in object file built for iOS Simulator, file '/path/to/app/build/ios/Debug-iphoneos/BoringSSL-GRPC.framework/BoringSSL-GRPC' for architecture arm64
`

When I remove the pre-compiled binary dependency from my Podfile (e.g. `pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git', :tag => '7.3.0'`) and run
```
rm -rf ~/Library/Developer/Xcode/DerivedData/
flutter clean
```
it builds fine and starts on my iOS Device.

Here is some more info which might be relevant to the issue:
In pubspec.yml I depend on `cloud_firestore: ^0.16.0`
macOS: 11.2.1 (Intel x86)
XCode: 12.4 (12D4e)
Flutter:
```
Flutter 1.22.6 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 9b2d32b605 (3 weeks ago) • 2021-01-22 14:36:39 -0800
Engine • revision 2f0af37152
Tools • Dart 2.10.5
```

I never really use Xcode, so it might be something really obvious I am missing. When trying to search for the error above I only found people having the reverse Issue, e.g. trying to build for Simulator and failing.

The obvious workaround for now is not to include the pre-compiled binaries, which is fine but I would prefer a more permanent solution which still uses the binaries and have the build time improvements.

Any help would be much appreciated and if I can provide more info on my setup I will gladly do so. Thanks!

**Comment by mikehardy** [comment]
Have you tried with the most current version of firebase-ios-sdk / firebaes-ios-sdk-frameworks? it's 7.6.0 now

I know they have been doing lots of work to get Apple SIlicon stuff going in there, and errors related to "arm64" between ios and ios simulator are usually related to that these days.

Looking at the difference between the 2 releases:

https://github.com/invertase/firestore-ios-sdk-frameworks/tree/7.6.0/FirebaseFirestore/BoringSSL-GRPC.xcframework

https://github.com/invertase/firestore-ios-sdk-frameworks/tree/7.3.0/FirebaseFirestore/BoringSSL-GRPC.xcframework

One thing that is curious is that your error message says "BoringSSL-GRPC.framework", not "...xcframework". Why is BoringSSL coming in as a .framework (which will have this issue! it's the old style) instead of as the .xcframework from this release? :thinking: 

...they both seem to be including an .xcframework in a way that I'd expect. I wonder if your Podfile has some sort of arch excludes? There are a load of stackoverflow posts about Podfile workarounds to include or exclude different architectures when faced with this sort of message and most of them should not be needed anymore, but you might check them out.

**Comment by maximilianmaihoefner** [comment]
Thanks for your quick response @mikehardy!

Since it states in the Readme that you should use the version-tag matching the version of the FirebaseSDK that is by FlutterFire I didn't try to update to the newest version. Doing so just ends with a dependency resolution conflict:
```
    CocoaPods could not find compatible versions for pod "FirebaseFirestore":
      In Podfile:
        FirebaseFirestore (from `https://github.com/invertase/firestore-ios-sdk-frameworks.git`, tag `7.6.0`)

        cloud_firestore (from `.symlinks/plugins/cloud_firestore/ios`) was resolved to 0.16.0, which depends on
          Firebase/Firestore (= 7.3.0) was resolved to 7.3.0, which depends on
            FirebaseFirestore (~> 7.3.0)
```

Coincidentally, the current version also told me that it needs a newer version of cocoapods, after upgrading cocoapods and downgrading the pre-compiled binaries it now builds and launches on both the Simulator and a real Device.

So I guess the lesson for today is, that I should check my cocoapods version from time to time and see if there are updates for it.

Thanks for your help!

**Comment by mikehardy** [comment]
Yeah - all the Apple Silicon stuff has forced an upgrade to cocoapods 1.10+ in order to handle the exact .framework vs .xcframework thing that enables the extra slices to go in the library binaries needed for arm64 simulators to not conflict with arm64 real devices. It's kind of a mess but all sorted once .xcframeworks are in use everywhere

It is true that you do not want to mix and match things however there is not reason FlutterFire can't move from 7.3.0 to 7.6.0 - I'm the one that originally forward-ported FlutterFire to the 7.x firebase-ios-sdk series and I track those releases pretty closely as I maintain react-native-firebase too, it's already moving to 7.6.0. So if you wanted to try an override there (it's possible by putting it in your Podfile - there's a variable for it that FlutterFire will recognize) you could do that

However! There is no reason to do that if you're not affected by a bug fixed in 7.4.0/7.5.0/7.6.0 either and if it's working, maybe leave it be :-). Cheers
